### DISCUSSION TITLE
How to connect to a IPsec/L2TP server in github action

### DISCUSSION THREAD

**Discussion by XinmengLi2020** [community/community#31592] [Actions]
In my action flow, in order to connect to a database, it is necessary to connect to a  IPsec/L2TP VPN.
It is straightforward using Mac and Ubuntu Gui, whereas, not so in the GitHub action!

There are recommended solutions e.g. [L2TP Connection Client on ubuntu 18.04 Server](https://askubuntu.com/questions/1167283/l2tp-connection-client-on-ubuntu-18-04-server)

However, as people comment also in the post above, there are errors like "Error: Connection activation failed: Could not find source connection"

Anyone has already explored a protocol for connecting an IPsec/L2TP server? Any feedback is appreciated~

**Comment by jfullstackdev** [comment]
if this is the same project we are referring to, then asking their by raising an issue might help you, as others will see it, with the same focus and interest

https://github.com/hwdsl2/docker-ipsec-vpn-server

https://github.com/hwdsl2/setup-ipsec-vpn

**Reply by XinmengLi2020** [reply]
Hi @jdevstatic, thank you very much for your quick reply.
The project you referred to is to build a VPN server. There might be some guidance about set a VPN client, but mainly using GUI, e.g. https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/docs/clients.md

My problem should be simpler to solve, which is to connect an existing server using **nmcli** command directly. 
Some commands: 
```
sudo add-apt-repository ppa:nm-l2tp/network-manager-l2tp 
sudo apt-get install network-manager-l2tp

nmcli connection add connection.id [vpnName] con-name [vpnName] type VPN vpn-type l2tp ifname -- connection.autoconnect no ipv4.method auto vpn.data "gateway = [ipv4], ipsec-enabled = yes, ipsec-psk = 0s"$(base64 <<<'[PSK]' | rev | cut -c2- | rev)"=, mru = 1400, mtu = 1400, password-flags = 0, refuse-chap = yes, refuse-mschap = yes, refuse-pap = yes, require-mppe = yes, user = [user]" vpn.secrets password=[user-password]

- [vpnName] = The name of your connection
- [ipv4] = ip of the l2tp/ipsec server
- [PSK] = pre shared key from the l2tp/ipsec server
- [user] = user name to connect to
- [user-password] = password of the user to connect 

To show generated file: nmcli c show id [vpnName] 
To start the VPN from cli: nmcli c up [vpnName] 
To stop the VPN from cli: nmcli c down [vpnName] 
```
